tests/functional/lifecycle/LifecycleTask.js

```diff
@@ -174,6 +248,13 @@ class S3Helper {
                 if (!this._scenario[scenarioNumber]) {
                     return next(new Error('incorrect scenario number'));
                 }
+                // TODO: refactor
+                if (scenarioNumber === 2 || scenarioNumber === 3) {
```

Comment:
Instead of numbering scenarios, you might just as well create separate functions to prepare each scenario, called from ad-hoc places in test code. Or if you want to benefit from some factorization and genericity, you could pass a params object instead of scenario number, including the scenario params that apply to the particular test.

Reply:
I've decided to keep use of `scenarioNumber` because I think it'd be nice to reuse names and tags for any scenario to build with, and since the only differences were whether to create a non-versioned bucket, or versioned bucket, or versioned bucket w/ delete markers, I now pass the function name I want to the buildScenario method. 

Depending on how tests turn out for other rules, I'm sure there will be opportunities to refactor these tests further. If future tests don't really utilize re-use of scenarios, I'll update and pass in the list of names directly like you mentioned